sys/include/crypto/chacha20poly1305.h

```diff
@@ -38,6 +38,13 @@ extern "C" {
 #define CHACHA20POLY1305_NONCE_BYTES    (12U)   /**< Nonce length in bytes */
 #define CHACHA20POLY1305_TAG_BYTES      (16U)   /**< Tag length in bytes */
 
+typedef struct {
+    uint32_t constant[4];
+    uint32_t key[8];
+    uint32_t counter[1];
+    uint32_t nonce[3];
+} chacha20_ctx_t;
```

Comment:
If you rely on those being laid out in memory as a consecutive array, you should mark the struct with [`__attribute (packed)__`](https://gcc.gnu.org/onlinedocs/gcc-4.1.2/gcc/Type-Attributes.html)